agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/propagator/AiLegacyPropagator.java

```diff
@@ -137,4 +121,28 @@ private static String aiExtractRootId(String parentId) {
 
         return parentId.substring(rootStart, rootEnd);
     }
+
+    private static final long INVALID_ID = 0;
+
+    // copied from io.opentelemetry.sdk.trace.RandomIdGenerator
+    private static String generateSpanId() {
+        long id;
+        ThreadLocalRandom random = ThreadLocalRandom.current();
+        do {
```

Comment:
can you try `nextLong(1, long uppperBound)` so that i will never hit 0? and get rid off the while loop?

Reply:
that would not include any negative numbers, this is how it's implemented in OTel, so I think good to keep same implementation